### DISCUSSION TITLE
Double rust-analyzer instances after updating

### DISCUSSION THREAD

**Discussion by IAMSolaara** [LazyVim/LazyVim#3825] [Q&A] [answered]
Hello, I'm having this issue where after I updated my lazyvim config I get two different instances of rust-analyzer (one of which called rust_analyzer)
<img width="508" alt="image" src="https://github.com/LazyVim/LazyVim/assets/17319563/1dd203ed-03e1-428c-8bf4-5205ce86f08b">
My statusbar also has duplicate function names:
<img width="459" alt="image" src="https://github.com/LazyVim/LazyVim/assets/17319563/0f538f35-18c1-40ad-b25f-97830da19f61">

Often times this also gives me duplicate type hints after variables/parameters
<img width="549" alt="image" src="https://github.com/LazyVim/LazyVim/assets/17319563/42308bed-e418-46fd-98d8-b0636e27dce3">
<img width="421" alt="image" src="https://github.com/LazyVim/LazyVim/assets/17319563/3777e8e4-15d5-4f7b-b611-3d23bc8ddc27">

Does anyone have any clue on why this is happening?
I have the rust extras enabled: (this is a link to my dotfiles repo so you can see what I have enabled)
https://github.com/IAMSolaara/dotfilesNVIM/blob/f48e859ea7891aeee82390d12560d36122f24462/lua/config/lazy.lua#L31

This is my version of neovim:
<img width="412" alt="image" src="https://github.com/LazyVim/LazyVim/assets/17319563/5b5f774d-9b13-4b48-bedd-d9ea21cb5e02">
I'm on macOS 14.5 arm64.

Thanks in advance :D

**Comment by dpetka2001** [comment, marked as answer]
Your best bet is to test against a default LazyVim installation with only Rust Extra enabled and see if you observe the same behavior. If you don't then this is most likely some user configuration issue. You might also want to make sure you only have a single instance of `rust_analyzer` installed in your `$PATH`.

The double statusline function names is because you use `outline`. This gives symbols in the statusbar as well as LazyVim's default `trouble` plugin. You can set `vim.g.trouble_lualine = false` in your `options.lua` file to disable `trouble`'s symbols in the statusline since you use `outline` as well.

**Reply by IAMSolaara** [reply]
I ended up removing outline as I don't even really use it all that much.

As for rust_analyzer, I can only find rust-analyzer (with the dash,not underscore) in my machine
<img width="616" alt="image" src="https://github.com/LazyVim/LazyVim/assets/17319563/1420a65f-ed02-4df9-a955-0b8d33ffecb4">, so I tried spinning up a fedora container, installed neovim, git and used rustup to install a stable toolchain, then `rustup component add rust-analyzer` to install rust-analyzer. Then installed a fresh version of lazyvim on it and the problem doesn't occur.

Using LspInfo I can see the two attached LSP clients:
<img width="1300" alt="Screenshot 2024-06-27 at 11 02 17" src="https://github.com/LazyVim/LazyVim/assets/17319563/684b37d5-6e11-42b6-8161-cb953c53ef31">
Meanwhile in the docker container only one comes up:
<img width="709" alt="Screenshot 2024-06-27 at 11 26 58" src="https://github.com/LazyVim/LazyVim/assets/17319563/2a726dce-fba7-4572-949e-004155b19885">


I'll look into this more (and probably reinstall my config, its been around for a good while so a reinstall could do me some good) :/

**Reply by IAMSolaara** [reply]
Update: I got this fixed.
I removed rust-analyzer from mason, used `rustup component add rust-analyzer` from my host (which I didn't do before), took out the rust extras, did a Lazy sync, then readded them back, another lazy sync and it works as intended now.
I have no idea why this has happened tbh, but I guess it has to do with some manual fiddling I did when I first installed Lazyvim?
Anyway, thanks for the help :D

**Comment by amrro** [comment]
What I found out, while enabling rust extra plugins, is that rustaceanvim plugin [doesn't automatically detect `rust_analyzer`](https://github.com/mrcjkb/rustaceanvim/blob/af4de6a35d128ce71c75a9a7846bf089aea76f50/doc/mason.txt#L4-L8) from mason. 

by adding the following [LazyVim config](https://github.com/mrcjkb/rustaceanvim/blob/af4de6a35d128ce71c75a9a7846bf089aea76f50/doc/mason.txt#L40-L53) fixed it for me and now I'm having a single `rust_analyzer` client attached to buffer.

I'm new to neovim but I feel like this is worth mentioning!

**Comment by girish-48** [comment]
as mentioned in the rustaceanvim plugin, its better to not install rust_analyzer through mason, I had this same problem and running `brew install rust-analyzer` worked for me

**Comment by mariuscozma11** [comment]
For some reason, pressing u(update) inside the Mason interface hovering on rust-analyzer worked for me.
